server/src/domain/asset/asset.service.ts

```diff
@@ -2,12 +2,41 @@ import { Inject } from '@nestjs/common';
 import { AuthUserDto } from '../auth';
 import { IAssetRepository } from './asset.repository';
 import { MapMarkerDto } from './dto/map-marker.dto';
-import { MapMarkerResponseDto } from './response-dto';
+import { MapMarkerResponseDto, mapAsset } from './response-dto';
+import { MemoryLaneResponseDto } from './response-dto/memory-lane-response.dto';
+import { DateTime } from 'luxon';
 
 export class AssetService {
   constructor(@Inject(IAssetRepository) private assetRepository: IAssetRepository) {}
 
   getMapMarkers(authUser: AuthUserDto, options: MapMarkerDto): Promise<MapMarkerResponseDto[]> {
     return this.assetRepository.getMapMarkers(authUser.id, options);
   }
+
+  async getMemoryLane(authUser: AuthUserDto, timezone: string): Promise<MemoryLaneResponseDto[]> {
+    const result: MemoryLaneResponseDto[] = [];
+
+    const luxonDate = DateTime.fromISO(new Date().toISOString(), { zone: timezone });
+    const today = new Date(luxonDate.year, luxonDate.month - 1, luxonDate.day);
+
+    const years = Array.from({ length: 30 }, (_, i) => {
```

Comment:
Maybe rather than a static number like 30, there should be a config which sets an upper limit?

Also, might it be more efficient to query the database for all distinct years? Would therefore be a single query followed by however many more are required, or just one query if you want to go crazy, instead of 30 always.

Something like:

```sql
SELECT DATE_PART('year', date_creation::date) AS year, count(*)
FROM assets
WHERE EXTRACT(MONTH FROM fileCreatedAt) = 2 AND EXTRACT(MONTH FROM fileCreatedAt) = 15;
GROUP BY year;
```

I haven't tested it, but theoretically it would produce something like:

```ts
const result = {
    2023: [{ fileCreatedAt: '2023-02-15', ... }],
    2000: [{ fileCreatedAt: '2000-02-15', ... }, { fileCreatedAt: '2000-02-15', ... }],
};
```